File: app/src/main/java/com/cxwl/shawn/wuzhishan/decision/adapter/CityFragmentAdapter.java
package com.cxwl.shawn.wuzhishan.decision.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.cxwl.shawn.wuzhishan.decision.R;
import com.cxwl.shawn.wuzhishan.decision.dto.CityDto;

import java.util.List;

/**
 * 城市列表
 */
public class CityFragmentAdapter extends BaseAdapter {
	
	private Context mContext;
	private LayoutInflater mInflater;
	private List<CityDto> mArrayList;
	
	private final class ViewHolder{
		TextView tvName;
	}
	
	public CityFragmentAdapter(Context context, List<CityDto> mArrayList) {
		mContext = context;
		this.mArrayList = mArrayList;
		mInflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
	}

	@Override
	public int getCount() {
		return mArrayList.size();
	}

	@Override
	public Object getItem(int position) {
		return position;
	}

	@Override
	public long getItemId(int position) {
		return position;
	}

	@Override
	public View getView(int position, View convertView, ViewGroup parent) {
		ViewHolder mHolder;
		if (convertView == null) {
			convertView = mInflater.inflate(R.layout.adapter_city_nation, null);
			mHolder = new ViewHolder();
			mHolder.tvName = convertView.findViewById(R.id.tvName);
			convertView.setTag(mHolder);
		}else {
			mHolder = (ViewHolder) convertView.getTag();
		}
		
		CityDto dto = mArrayList.get(position);

		mHolder.tvName.setText(dto.disName);
		
		return convertView;
	}

}

File: app/src/main/java/com/cxwl/shawn/wuzhishan/decision/adapter/CityFragmentAdapter.kt
package com.cxwl.shawn.wuzhishan.decision.adapter;

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.cxwl.shawn.wuzhishan.decision.R
import com.cxwl.shawn.wuzhishan.decision.dto.CityDto

/**
 * 城市列表
 */
class CityFragmentAdapter constructor(context: Context?, private val mArrayList: ArrayList<CityDto>) : BaseAdapter() {
	
	private val mInflater = context!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

	class ViewHolder{
		var tvName : TextView? = null
	}

	override fun getItem(p0: Int): Any {
		return p0
	}

	override fun getItemId(p0: Int): Long {
		return p0.toLong()
	}

	override fun getCount(): Int {
		return mArrayList.size
	}

	override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
		val mHolder : ViewHolder
		val view : View
		if (convertView == null) {
			view = mInflater.inflate(R.layout.adapter_city_nation, null)
			mHolder = ViewHolder()
			mHolder.tvName = view.findViewById(R.id.tvName)
			view.tag = mHolder
		}else {
			view = convertView
			mHolder = view.tag as ViewHolder
		}

		val dto = mArrayList[position]
		mHolder.tvName!!.text = dto.disName

		return view
	}

}
